locallibrary/catalog/models.py

```diff
@@ -0,0 +1,98 @@
+from django.db import models
+import uuid
+from django.urls import reverse
+# Create your models here.
+class MyModelName(models.Model):
+    """A typical class defining a model, derived from the Model class."""
+
+    # Fields
+    my_field_name = models.CharField(max_length=20, help_text='Enter field documentation')
+    ...
+
+    # Metadata
+    class Meta:
+        ordering = ['-my_field_name']
+
+    # Methods
+    def get_absolute_url(self):
+        """Returns the url to access a particular instance of MyModelName."""
+        return reverse('model-detail-view', args=[str(self.id)])
+
+    def __str__(self):
+        """String for representing the MyModelName object (in Admin site etc.)."""
+        return self.my_field_name
+class Genre(models.Model):
+    """Model representing a book genre."""
+    name = models.CharField(max_length=200, help_text='Enter a book genre (e.g. Science Fiction)')
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+    title = models.CharField(max_length=200)
+
+    # Foreign Key used because book can only have one author, but authors can have multiple books
+    # Author as a string rather than object because it hasn't been declared yet in the file
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
```

Comment:
em tìm hiểu thêm option `on_delete` có những giá trị nào ? tìm hiểu về các giá trị này

Reply:
Có 6 hành động có thể thực hiện khi sự kiện đó xảy ra:

CASCADE: Khi đối tượng được tham chiếu bị xóa, cũng xóa các đối tượng có tham chiếu đến nó (Khi bạn xóa một bài đăng trên blog chẳng hạn, bạn cũng có thể muốn xóa bình luận). Tương đương SQL : CASCADE.
PROTECT: Cấm xóa đối tượng được tham chiếu. Để xóa nó, bạn sẽ phải xóa tất cả các đối tượng tham chiếu thủ công. Tương đương SQL : RESTRICT.
SET_NULL: Đặt tham chiếu thành NULL (yêu cầu trường là nullable). Chẳng hạn, khi bạn xóa một Người dùng, bạn có thể muốn giữ các bình luận mà anh ấy đã đăng trên các bài đăng trên blog, nhưng nói rằng nó đã được đăng bởi một người dùng ẩn danh (hoặc đã xóa). Tương đương SQL : SET NULL.
SET_DEFAULT: Đặt giá trị mặc định. Tương đương SQL : SET DEFAULT.
SET(...): Đặt một giá trị nhất định. Đây không phải là một phần của tiêu chuẩn SQL và hoàn toàn được xử lý bởi Django.
DO_NOTHING: Có lẽ là một ý tưởng rất tồi vì điều này sẽ tạo ra các vấn đề toàn vẹn trong cơ sở dữ liệu của bạn (tham chiếu một đối tượng thực sự không tồn tại). Tương đương SQL : NO ACTION.